src/main/java/gregtech/common/covers/detector/CoverDetectorEnergyAdvanced.java

```diff
@@ -66,166 +61,222 @@ public EnumActionResult onScrewdriverClick(EntityPlayer playerIn, EnumHand hand,
-        try {
-            long c = Long.parseLong(val);
-            maxEU = Math.max(minEU + 1, c);
-        } catch (NumberFormatException e) {
-            GTLog.logger.warn(e);
-            this.maxEU = Math.max(minEU + 1, 2048);
+        long parsedValue;
+        long maxUpperLimit;
+
+        if (usePercent) {
+            parsedValue = GTUtility.tryParseLong(val, DEFAULT_MAX_PERCENT);
+            maxUpperLimit = 100;
+        } else {
+            parsedValue = GTUtility.tryParseLong(val, DEFAULT_MAX_EU);
+            maxUpperLimit = Long.MAX_VALUE;
         }
+
+        this.maxValue = Math.max(this.minValue + 1, Math.min(parsedValue, maxUpperLimit));
     }
 
     private boolean isInverted(){
-        return this.inverted;
+        return this.isInverted;
     }
 
     private void setInverted(boolean b){
-        this.inverted = b;
+        this.isInverted = b;
     }
 
-    @Override
-    public boolean canConnectRedstone() {
-        return true;
+    private boolean isUsePercent(){
+        return this.usePercent;
+    }
+
+    private void setUsePercent(boolean b){
+        this.usePercent =  b;
+        int length;
+
+        if (this.usePercent){ // using percent
+            this.minValue = DEFAULT_MIN_PERCENT;
+            this.maxValue = DEFAULT_MAX_PERCENT;
+            length = 3;
+        } else { // using discrete EU
+            this.minValue = DEFAULT_MIN_EU;
+            this.maxValue = DEFAULT_MAX_EU;
+            length = 19;
+        }
+
+        // update widgets
+        updateSyncedWidgets(length);
+    }
+
+    private void updateSyncedWidgets(int length) {
+        for (Widget widget : this.widgetsToUpdate.widgets) {
+            if (widget instanceof TextFieldWidget2) {
+                ((TextFieldWidget2) widget).setPostFix(null); // clear postfix
+                ((TextFieldWidget2) widget).setPostFix(this.getPostFix());
+                ((TextFieldWidget2) widget).setMaxLength(length);
+            }
+        }
+    }
+
+    private void updateSyncedWidgets() {
+        for (Widget widget : this.widgetsToUpdate.widgets) {
+            if (widget instanceof TextFieldWidget2) {
+                ((TextFieldWidget2) widget).setPostFix(null); // clear postfix
+                ((TextFieldWidget2) widget).setPostFix(this.getPostFix());
+            }
+        }
+    }
+
+    private String getPostFix(){
+        return this.usePercent ? " %" : " EU";
     }
 
     @Nonnull
     @Override
     public NBTTagCompound writeToNBT(@Nonnull NBTTagCompound tagCompound) {
         super.writeToNBT(tagCompound);
-        tagCompound.setLong("maxEU", this.maxEU);
-        tagCompound.setLong("minEU", this.minEU);
+        tagCompound.setLong("maxEU", this.maxValue);
+        tagCompound.setLong("minEU", this.minValue);
         tagCompound.setInteger("outputAmount", this.outputAmount);
-        tagCompound.setBoolean("inverted", this.inverted);
-        tagCompound.setBoolean("isEnabled", this.isEnabled);
```

Comment:
Given this changes it should go out in minor update not revision and it should be documented - as this may blow up bases...

Reply:
I accidentally removed the read and write of `isInverted`, so that has been added back in. Removing `isEnabled` is probably fine, I think